### PR TITLE
Temporal colouring of the project_status page's statuses

### LINKED ISSUE #225: Project status last updated
Add in the project/plate status a `last_update_date` field which will be the most recent change to the sample with this status.
In the project status web page background colour the cells where the `last_update_date` field is more than 1, 2  and 4 weeks.
this will not apply to the finished and removed status.

### PR DESCRIPTION
**Pull request by lukbut**
Closes #225 

From the issue:

"Add in the project/plate status a last_update_date field which will be the most recent change to the sample with this status.
In the project status web page background colour the cells where the last_update_date field is more than 1, 2 and 4 weeks.
this will not apply to the finished and removed status."

### CODE REVIEW

File: rest_api/limsdb/queries/data_models.py [hunk 1]
```diff
@@ -247,7 +253,8 @@ def to_json(self):
             'library_type': self.library_types,
             'species': self.species,
             'required_yield': self.required_yield,
-            'required_coverage': self.coverage
+            'required_coverage': self.coverage,
```
Comment (lukbut) [inline rest_api/limsdb/queries/data_models.py]:
```python
status, dates = self.samples_per_status()

ret = {
....
 'sample_per_status_date' : dates
}
ret.update(status)

```

Comment (tcezard) [reply rest_api/limsdb/queries/data_models.py]:
```python
status: {
    delivery:{
        samples: []
        last_modified_date:
    }
}
```

File: reporting_app/static/column_formatter.js [hunk 2]
```diff
@@ -156,6 +185,32 @@ function count_entities_fmt(data, fmt){
...
+ */
+    // Checking time elapsed of the status' max date
+    status_date = new Date(cell_data['last_modified_date']);
+    // Creating fixed date variable to compare against
+    week_ago = new Date();
+    week_ago.setDate(week_ago.getDate() - 7)
+    two_weeks_ago = new Date();
+    two_weeks_ago.setDate(week_ago.getDate() - 14)
+    four_weeks_ago = new Date();
+    four_weeks_ago.setDate(week_ago.getDate() - 28)
```
Comment (tcezard) [inline reporting_app/static/column_formatter.js]: This section could be simplify using moment.js 
```javascript
    days_since_last_change = moment().diff(cell_data['last_modified_date'], 'days')
```
The compare `days_since_last_change` with the thredholds

File: reporting_app/static/datatable.js [hunk 3]
```diff
@@ -348,7 +348,13 @@ var sum_row_per_column = function( row, data, start, end, display ) {
                 .cells( null, this.index(), { page: 'current'} )
                 .data()
                 .reduce(function (a, b) {
-                    if (b.constructor === Array){
+                    /* This is a hardcoded workaround, which is expected to be in place until a need for a more generalised
+                       solution arises.
+                     */
+                    if (b instanceof Object){
+                        return b.samples.length;
```
Comment (tcezard) [inline reporting_app/static/datatable.js]: :sob: :sob: :sob:

Comment (lukbut) [reply reporting_app/static/datatable.js]: 💔

File: reporting_app/static/column_formatter.js (hunk 2 above)
Comment (lukbut) [reply reporting_app/static/column_formatter.js]: Do we have moment.js installed?

Comment (lukbut) [reply reporting_app/static/column_formatter.js]: Great tip, worked like a charm. Thanks.

File: reporting_app/static/datatable.js (hunk 3 above)
Comment (mwhamgenomics) [reply reporting_app/static/datatable.js]: FeelsBadMan

**Pull request merged**